registry/storage/registry.go

```diff
@@ -12,28 +12,60 @@ import (
 // package. All instances should descend from this object.
 type registry struct {
 	blobStore                   *blobStore
-	blobServer                  distribution.BlobServer
-	statter                     distribution.BlobStatter // global statter service.
+	blobServer                  *blobServer
```

Comment:
I would be more comfortable not using the interfaces here. Without the interfaces, the compiler ensures that the right types are used. Using an interface and type assertions, we'd have a runtime problem if someone tried to put a different type in one of these fields.